diff --git a/Core/Inc/stm32_assert.h b/Core/Inc/stm32_assert.h
index ca09699..cc7a50a 100644
--- a/Core/Inc/stm32_assert.h
+++ b/Core/Inc/stm32_assert.h
@@ -19,6 +19,7 @@
 /* Define to prevent recursive inclusion -------------------------------------*/
 #ifndef __STM32_ASSERT_H
 #define __STM32_ASSERT_H
+#include <stdint.h>
 
 #ifdef __cplusplus
  extern "C" {
diff --git a/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f401xe.h b/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f401xe.h
index dd54cf1..4f7f45d 100644
--- a/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f401xe.h
+++ b/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f401xe.h
@@ -8284,7 +8284,7 @@ typedef struct
                                        ((INSTANCE) == I2C3))
 
 /******************************* SMBUS Instances ******************************/
-#define IS_SMBUS_ALL_INSTANCE         IS_I2C_ALL_INSTANCE
+#define IS_SMBUS_ALL_INSTANCE(x)         IS_I2C_ALL_INSTANCE(x)
 
 /******************************** I2S Instances *******************************/
 
@@ -8295,7 +8295,7 @@ typedef struct
 #define IS_I2S_EXT_ALL_INSTANCE(INSTANCE) (((INSTANCE) == I2S2ext)|| \
                                            ((INSTANCE) == I2S3ext))
 /* Legacy Defines */
-#define IS_I2S_ALL_INSTANCE_EXT    IS_I2S_EXT_ALL_INSTANCE
+#define IS_I2S_ALL_INSTANCE_EXT(x)    IS_I2S_EXT_ALL_INSTANCE(x)
 
 
 /****************************** RTC Instances *********************************/
@@ -8559,14 +8559,14 @@ typedef struct
                                                ((INSTANCE) == USART6))
 
 /* Legacy defines */
-#define IS_UART_INSTANCE          IS_UART_HALFDUPLEX_INSTANCE
+#define IS_UART_INSTANCE(x)          IS_UART_HALFDUPLEX_INSTANCE(x)
 
 /****************** UART Instances : Hardware Flow control ********************/
 #define IS_UART_HWFLOW_INSTANCE(INSTANCE) (((INSTANCE) == USART1) || \
                                            ((INSTANCE) == USART2) || \
                                            ((INSTANCE) == USART6))
 /******************** UART Instances : LIN mode **********************/
-#define IS_UART_LIN_INSTANCE          IS_UART_HALFDUPLEX_INSTANCE
+#define IS_UART_LIN_INSTANCE(x)          IS_UART_HALFDUPLEX_INSTANCE(x)
 
 /********************* UART Instances : Smart card mode ***********************/
 #define IS_SMARTCARD_INSTANCE(INSTANCE) (((INSTANCE) == USART1) || \
diff --git a/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h b/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h
index 052b285..3114768 100644
--- a/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h
+++ b/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h
@@ -8,8 +8,8 @@
   *          is using in the C source code, usually in main.c. This file contains:
   *           - Configuration section that allows to select:
   *              - The STM32F4xx device used in the target application
-  *              - To use or not the peripheral’s drivers in application code(i.e. 
-  *                code will be based on direct access to peripheral’s registers 
+  *              - To use or not the peripheral's drivers in application code(i.e. 
+  *                code will be based on direct access to peripheral's registers 
   *                rather than drivers API), this option is controlled by 
   *                "#define USE_HAL_DRIVER"
   *  
diff --git a/Drivers/CMSIS/Device/ST/STM32F4xx/Include/system_stm32f4xx.h b/Drivers/CMSIS/Device/ST/STM32F4xx/Include/system_stm32f4xx.h
index 8087ba4..5f389a7 100644
--- a/Drivers/CMSIS/Device/ST/STM32F4xx/Include/system_stm32f4xx.h
+++ b/Drivers/CMSIS/Device/ST/STM32F4xx/Include/system_stm32f4xx.h
@@ -46,6 +46,7 @@
   */
 #ifndef __SYSTEM_STM32F4XX_H
 #define __SYSTEM_STM32F4XX_H
+#include <stdint.h>
 
 #ifdef __cplusplus
  extern "C" {
